src/map/packet_system.cpp

```diff
@@ -6137,12 +6143,17 @@ void SmallPacket0x111(map_session_data_t* session, CCharEntity* PChar, CBasicPac
 
 void SmallPacket0x112(map_session_data_t* session, CCharEntity* PChar, CBasicPacket data)
 {
-    // Send spark updates + current RoE quests
+    // Send spark updates
     PChar->pushPacket(new CRoeSparkUpdatePacket(PChar));
-    PChar->pushPacket(new CRoeUpdatePacket(PChar));
-    // 4-part Eminence Completion bitmap
-    for(int i = 0; i < 4; i++)
-        PChar->pushPacket(new CRoeQuestLogPacket(PChar, i));
+
+    if (roeutils::RoeSystem.RoeEnabled)
+    {
+        // Current RoE quests
+        PChar->pushPacket(new CRoeUpdatePacket(PChar));
+        // 4-part Eminence Completion bitmap
+        for(int i = 0; i < 4; i++)
+            PChar->pushPacket(new CRoeQuestLogPacket(PChar, i));
```

Comment:
Space after for
Brackets around the statement

Reply:
This line actually pre-dates this commit. I didn't modify it. and both lines really represent a singular command, as the questLog must **always** be as a set of 4.

Reply:
Are we averse to having it all on one line?

Reply:
The end result I meant was:
```cpp
for (int i = 0; i < 4; i++)
{
    PChar->pushPacket(new CRoeQuestLogPacket(PChar, i));
}
```

I think the only thing I like for single line ifs are returns.

Maybe it wouldn't be such a bad thing to just have these 4 lines without a loop?
```cpp
PChar->pushPacket(new CRoeQuestLogPacket(PChar, 0));
PChar->pushPacket(new CRoeQuestLogPacket(PChar, 1));
PChar->pushPacket(new CRoeQuestLogPacket(PChar, 2));
PChar->pushPacket(new CRoeQuestLogPacket(PChar, 3));
```

Reply:
...I will put the brackets... I just hate how bloated single-line braced sections look. :unamused: 

My preference would to avoid bracing would be:
```c++
for (int i = 0; i < 4; i++) PChar->pushPacket(new CRoeQuestLogPacket(PChar, i));
```

But I didn't want to unroll it, since the compiler will be doing that anyway, and I suspected it's possible that SE may expand the record system in the future; there's clear evidence that's why this multi-packet situation already exists.